data-raw/create_sdtms_data.R

```diff
@@ -0,0 +1,156 @@
+#' This script generates R documentation files for datasets in the 'pharmaversesdtm' package.
+#'
+#' It reads metadata from a JSON file ('inst/extdata/sdtms-specs.json') and uses that information to create R documentation files
+#' for each dataset. The script utilizes helper functions to handle metadata retrieval and format the documentation properly.
+#' The generated documentation files are saved in the 'R' directory and should not be manually edited, as they are subject to automatic regeneration.
+
+# Load the necessary library for JSON
+library(jsonlite)
+
+# Load the JSON file with metadata
+specs <- fromJSON("inst/extdata/sdtms-specs.json")$`sdtms-specs`
+
+# Helper function to get attribute for a column, with default as "undocumented field"
+get_attr <- function(data, col_name) {
+  att <- attr(data[[col_name]], "label")
+  if (is.null(att) || att == "null") {
+    att <- "undocumented field"
+  }
+  return(att)
+}
+
+# Helper function to check the length of a string and break it into lines if it exceeds 80 characters
+check_and_break_line <- function(text, max_length = 80) {
+  if (nchar(text) > max_length) {
+    # Split the text into chunks of at most max_length characters
+    split_text <- strsplit(text, "(?<=.{1," , max_length, "})(\\s|\\n)", perl = TRUE)[[1]]
+    # Join the chunks with newline character
+    text <- paste(split_text, collapse = "\n")
+  }
+  return(text)
+}
+
+# Function to generate an HTML hyperlink
+generate_hyperlink <- function(url, link_text = "") {
+  if (!nzchar(url)) {
+    stop("URL cannot be empty.")
+  }
+  # Generate and return the HTML hyperlink
+  paste0('<a href="', url, '" target="_blank">', "Access the source of the ", link_text,' dataset.</a>')
+}
+
+# Define the `write_doc` function, now with `dataset_description`
+write_doc <-
+  function(data,
+           dataset_name,
+           dataset_label = "No label available",
+           dataset_description = "No description available",
+           dataset_author = NULL,
+           dataset_source = "No source available") {
+    # Check and adjust lengths of metadata fields
+    dataset_label <- check_and_break_line(dataset_label)
+    dataset_description <- check_and_break_line(dataset_description)
+    dataset_source <- check_and_break_line(generate_hyperlink(dataset_source, dataset_label))
+    if (!is.null(dataset_author)) {
+      dataset_author <- check_and_break_line(dataset_author)
+    }
+
+    # Create documentation for the current dataset
+    doc_string <- paste(
+      sprintf("# This file is automatically generated by data-raw/create_sdtms_data.R."),
+      sprintf("# For updating it please edit inst/extdata/sdtms-specs.json and rerun create_sdtms_data.R."),
+      sprintf("# Manual edits are not recommended, as changes may be overwritten."),
+      sprintf("#' %s", dataset_label),
```

Comment:
This will fail once `check_and_break_line()` works because only the first line starts with `#'` then.